From: Remi Collet <remi@remirepo.net>
Date: Mon, 24 Aug 2020 15:55:54 +0200
Subject: fix for PHP 8.0.0beta2

---
 zmq-1.1.3/zmq.c        | 2 ++
 zmq-1.1.3/zmq_device.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/zmq-1.1.3/zmq.c b/zmq-1.1.3/zmq.c
index 17dfb3e..1040cd1 100644
--- a/zmq-1.1.3/zmq.c
+++ b/zmq-1.1.3/zmq.c
@@ -621,7 +621,9 @@ zend_bool php_zmq_connect_callback(zval *socket, zend_fcall_info *fci, zend_fcal
 	fci->params         = params;
 	fci->param_count    = 2;
 	fci->retval         = &retval;
+#if PHP_VERSION_ID < 80000
 	fci->no_separation  = 1;
+#endif
 
 	if (zend_call_function(fci, fci_cache) == FAILURE) {
 		if (!EG(exception)) {
diff --git a/zmq-1.1.3/zmq_device.c b/zmq-1.1.3/zmq_device.c
index 534f966..1c6aa3e 100644
--- a/zmq-1.1.3/zmq_device.c
+++ b/zmq-1.1.3/zmq_device.c
@@ -53,7 +53,9 @@ zend_bool s_invoke_device_cb (php_zmq_device_cb_t *cb, uint64_t current_ts)
 	cb->fci.param_count = 1;
 
 	/* Call the cb */
+#if PHP_VERSION_ID < 80000
 	cb->fci.no_separation  = 1;
+#endif
 	cb->fci.retval         = &fc_retval;
 
 	if (zend_call_function(&(cb->fci), &(cb->fci_cache)) == FAILURE) {
