From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Tue, 1 Nov 2016 08:56:57 +0100
Subject: Skip version check by default

---
 imagick-3.7.0/imagick.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/imagick-3.7.0/imagick.c b/imagick-3.7.0/imagick.c
index 7733e2b..5606e7f 100644
--- a/imagick-3.7.0/imagick.c
+++ b/imagick-3.7.0/imagick.c
@@ -132,7 +132,7 @@ static void php_imagick_object_free_storage(IM_ZEND_OBJECT *object TSRMLS_DC)
 	}
 
 	if (intern->progress_monitor_name) {
-		//TODO - is this meant to be free by us? 
+		//TODO - is this meant to be free by us?
 		efree(intern->progress_monitor_name);
 	}
 
@@ -437,7 +437,7 @@ static zend_object_value php_imagickpixel_object_new_ex(zend_class_entry *class_
 	intern->initialized_via_iterator = 0;
 	zend_object_std_init(&intern->zo, class_type TSRMLS_CC);
 	object_properties_init(&intern->zo, class_type);
-	
+
 #if PHP_VERSION_ID >= 70000
 	intern->zo.handlers = &imagickpixel_object_handlers;
 
@@ -523,7 +523,7 @@ static zend_object_value php_imagickkernel_object_new(zend_class_entry *class_ty
 
 PHP_INI_BEGIN()
 	STD_PHP_INI_ENTRY("imagick.locale_fix", "0", PHP_INI_ALL, OnUpdateBool, locale_fix, zend_imagick_globals, imagick_globals)
-	STD_PHP_INI_ENTRY("imagick.skip_version_check", "0", PHP_INI_ALL, OnUpdateBool, skip_version_check, zend_imagick_globals, imagick_globals)
+	STD_PHP_INI_ENTRY("imagick.skip_version_check", "1", PHP_INI_ALL, OnUpdateBool, skip_version_check, zend_imagick_globals, imagick_globals)
 	STD_PHP_INI_ENTRY("imagick.progress_monitor", "0", PHP_INI_SYSTEM, OnUpdateBool, progress_monitor, zend_imagick_globals, imagick_globals)
 
 	STD_PHP_INI_ENTRY("imagick.set_single_thread", "1", PHP_INI_SYSTEM, OnUpdateBool, set_single_thread, zend_imagick_globals, imagick_globals)
@@ -535,7 +535,7 @@ static void php_imagick_init_globals(zend_imagick_globals *imagick_globals)
 {
 	imagick_globals->locale_fix = 0;
 	imagick_globals->progress_monitor = 0;
-	imagick_globals->skip_version_check = 0;
+	imagick_globals->skip_version_check = 1;
 	imagick_globals->set_single_thread = 1;
 	imagick_globals->allow_zero_dimension_images = 0;
 	// 10 is magick number, that seems to be enough.
@@ -726,7 +726,7 @@ static zval *php_imagick_read_property(zval *object, zval *member, int type, con
 	ret = std_hnd->has_property(object, member, 2 TSRMLS_CC);
 #else
 	ret = std_hnd->has_property(object, member, 2, key TSRMLS_CC);
-#endif	
+#endif
 
 	if (ret) {
 		std_hnd = zend_get_std_object_handlers();
@@ -1176,7 +1176,7 @@ PHP_MINFO_FUNCTION(imagick)
 #endif
 			IMAGICK_FREE_MAGICK_MEMORY(supported_formats[i]);
 		}
-		
+
 #if PHP_VERSION_ID >= 70000
 		smart_string_0(&formats);
 		php_info_print_table_row(2, "ImageMagick supported formats", formats.c);
