Index: llvm-toolchain-7-7/tools/llvm-shlib/CMakeLists.txt
===================================================================
--- llvm-toolchain-7-7.orig/tools/llvm-shlib/CMakeLists.txt
+++ llvm-toolchain-7-7/tools/llvm-shlib/CMakeLists.txt
@@ -40,6 +40,7 @@ set_property(TARGET LLVM PROPERTY VERSIO
 list(REMOVE_DUPLICATES LIB_NAMES)
 if(("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") OR (MINGW) OR (HAIKU)
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+   OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "GNU")
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "OpenBSD")
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "Fuchsia")
    OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "DragonFly")
