From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Tue, 30 May 2017 12:05:18 +0200
Subject: Use DEB_HOST_MULTIARCH in config.m4

---
 lua-1.1.0/config.m4 | 2 +-
 lua-2.0.7/config.m4 | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lua-1.1.0/config.m4 b/lua-1.1.0/config.m4
index 19341af..4c30bde 100644
--- a/lua-1.1.0/config.m4
+++ b/lua-1.1.0/config.m4
@@ -50,7 +50,7 @@ if test "$PHP_LUA" != "no"; then
     LUA_LIB_DIR=$PHP_LUA/$PHP_LIBDIR
   else
     AC_MSG_CHECKING(for lua library in default path)
-    for i in /usr/$PHP_LIBDIR /usr/lib /usr/lib64 /usr/lib/x86_64-linux-gnu; do
+    for i in /usr/$PHP_LIBDIR /usr/lib /usr/lib/${DEB_HOST_MULTIARCH}; do
       if test -r $i/${LUA_LIB_NAME}.${SHLIB_SUFFIX_NAME} -o -r $i/${LUA_LIB_NAME}.a; then
         LUA_LIB_DIR=$i
         AC_MSG_RESULT(found in $i)
diff --git a/lua-2.0.7/config.m4 b/lua-2.0.7/config.m4
index ab64009..27b95fe 100755
--- a/lua-2.0.7/config.m4
+++ b/lua-2.0.7/config.m4
@@ -49,7 +49,7 @@ if test "$PHP_LUA" != "no"; then
     LUA_LIB_DIR=$PHP_LUA/$PHP_LIBDIR
   else
     AC_MSG_CHECKING(for lua library in default path)
-    for i in /usr/$PHP_LIBDIR /usr/lib /usr/lib64 /usr/lib/x86_64-linux-gnu; do
+    for i in /usr/$PHP_LIBDIR /usr/lib /usr/lib/${DEB_HOST_MULTIARCH}; do
       if test -r $i/${LUA_LIB_NAME}.${SHLIB_SUFFIX_NAME} -o -r $i/${LUA_LIB_NAME}.a; then
         LUA_LIB_DIR=$i
         AC_MSG_RESULT(found in $i)
