Index: llvm-toolchain-7-7.0.1/include/llvm/MC/MCELFObjectWriter.h
===================================================================
--- llvm-toolchain-7-7.0.1.orig/include/llvm/MC/MCELFObjectWriter.h
+++ llvm-toolchain-7-7.0.1/include/llvm/MC/MCELFObjectWriter.h
@@ -75,6 +75,7 @@ public:
         return ELF::ELFOSABI_CLOUDABI;
       case Triple::PS4:
       case Triple::FreeBSD:
+      case Triple::kFreeBSD:
         return ELF::ELFOSABI_FREEBSD;
       default:
         return ELF::ELFOSABI_NONE;
