Index: llvm-toolchain-7-7/clang-tools-extra/clangd/CMakeLists.txt
===================================================================
--- llvm-toolchain-7-7.orig/clang-tools-extra/clangd/CMakeLists.txt
+++ llvm-toolchain-7-7/clang-tools-extra/clangd/CMakeLists.txt
@@ -7,6 +7,11 @@ if(NOT HAVE_CXX_ATOMICS64_WITHOUT_LIB)
   list(APPEND CLANGD_ATOMIC_LIB "atomic")
 endif()
 
+if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i386" OR
+   CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "i686")
+  list(APPEND CLANGD_ATOMIC_LIB "atomic")
+endif()
+
 add_clang_library(clangDaemon
   AST.cpp
   ClangdLSPServer.cpp
