Index: llvm-toolchain-3.9-3.9/clang/tools/scan-view/bin/scan-view
===================================================================
--- llvm-toolchain-3.9-3.9.orig/clang/tools/scan-view/bin/scan-view
+++ llvm-toolchain-3.9-3.9/clang/tools/scan-view/bin/scan-view
@@ -23,6 +23,7 @@ kDefaultPort = 8181
 kMaxPortsToTry = 100
 
 ###
+BASE_DIR = '/usr/share/clang/scan-view-7'
 
 
 def url_is_up(url):
@@ -61,7 +62,7 @@ def start_browser(port, options):
 
 def run(port, options, root):
     # Prefer to look relative to the installed binary
-    share = os.path.dirname(__file__) + "/../share/"
+    share = os.path.join(BASE_DIR, 'share')
     if not os.path.isdir(share):
         # Otherwise look relative to the source
         share = os.path.dirname(__file__) + "/../../scan-view/share"
