Description: create configure.local to set Debian defaults
Author: Bdale Garbee <bdale@gag.com>

--- /dev/null
+++ mdocml-1.13.3/configure.local
@@ -0,0 +1,13 @@
+HAVE_WCHAR=1
+OSNAME="Debian"
+PREFIX="/usr"
+MANDIR="${PREFIX}/share/man"
+EXAMPLEDIR="${PREFIX}/share/doc/mandoc"
+INSTALL_MAN="debian/install_mandoc"
+MANM_MAN="mandoc_man"           # default is "man"
+MANM_MDOC="mandoc_mdoc"         # default is "mdoc"
+MANM_ROFF="mandoc_roff"         # default is "roff"
+MANM_EQN="mandoc_eqn"           # default is "eqn"
+MANM_TBL="mandoc_tbl"           # default is "tbl"
+BINM_MAN=mman                   # default is "man"
+BUILD_DB=0
